src/lib/y2partitioner/widgets/blk_device_button.rb

```diff
@@ -0,0 +1,71 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cwm"
+
+module Y2Partitioner
+  module Widgets
+    # Base button class for modifying a block device (edit, resize, delete, etc)
+    class BlkDeviceButton < CWM::PushButton
+      # Constructor
+      # @param pager [CWM::TreePager]
+      # @param table [Y2Partitioner::Widgets::ConfigurableBlkDevicesTable]
+      # @param device [Y2Storage::BlkDevice]
+      def initialize(pager: nil, table: nil, device: nil)
+        textdomain "storage"
+
+        unless device || table
+          raise ArgumentError, "Please provide either a block device or a table with devices"
+        end
+
+        @pager = pager
+        @table = table
+        @device = device
+      end
+
+      # @macro seeAbstractWidget
+      def handle
+        if device.nil?
+          Yast::Popup.Error(_("No device selected"))
+          return nil
+        end
+
+        actions
+      end
+
+    protected
+
+      attr_reader :pager
```

Comment:
If you are not going to add documentation for each `attr_reader`, I would merge both of them.